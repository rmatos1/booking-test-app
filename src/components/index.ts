export * from "./bookingDescription";
export * from "./bookingForm";
export * from "./images";
export * from "./inputGroup";
export * from "./navBar";
export * from "./styledComponents";
export * from "./testComponent";
