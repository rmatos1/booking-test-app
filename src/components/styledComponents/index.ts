export * from './buttons';
export * from './formComponents';
export * from './layouts';
export * from './lists';
export * from './modalComponents';
export * from './texts';
