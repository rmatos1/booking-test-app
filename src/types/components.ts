import { ReactNode } from "react";

export interface IComponentWithChildren {
  children?: ReactNode;
}

export interface IBaseDrawer extends IComponentWithChildren {
  isDrawerVisible: boolean;
  onCloseDrawer: () => void;
}
