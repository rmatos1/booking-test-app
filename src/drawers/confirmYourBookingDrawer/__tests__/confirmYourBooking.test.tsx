import { describe, test } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';
import { TestComponent } from '../../../components';

import { ConfirmYourBookingDrawer } from '..';

const setup = (): JSX.Element => {
  return (
    <TestComponent>
      <ConfirmYourBookingDrawer />
    </TestComponent>
  );
};

describe('<ConfirmYourBookingDrawer />', () => {
  test('should render correctly', () => {
    const wrapper = render(setup());

    expect(wrapper).toMatchSnapshot();
  });

  test('should render the specified name', () => {
    const NAME = 'test';

    const wrapper = render(setup());

    const inputName = wrapper.getByTestId('input-name') as HTMLInputElement;

    fireEvent.change(inputName, { target: { value: NAME } });

    expect(inputName.value).toBe(NAME);
  });

  test('should render the specified email', () => {
    const EMAIL = 'test@test.com';

    const wrapper = render(setup());

    const inputEmail = wrapper.getByTestId('input-email') as HTMLInputElement;

    fireEvent.change(inputEmail, { target: { value: EMAIL } });

    expect(inputEmail.value).toBe(EMAIL);
  });
});
