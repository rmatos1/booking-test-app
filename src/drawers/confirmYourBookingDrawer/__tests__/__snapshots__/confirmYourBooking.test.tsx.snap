// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmYourBookingDrawer /> should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="sc-idOjMB kWZquY"
      />
      <div
        class="sc-cxFKTC hsKcGb"
        data-testid="base-drawer"
      >
        <button
          class="sc-dmlpXa gYemGx"
          data-testid="close-button"
        >
          X
        </button>
        <form
          class="sc-crrtmM XATYL"
        >
          <h3
            class="sc-dQpIV fPWcU"
          >
            Confirm Your Booking
          </h3>
          <div
            class="sc-jSguLX klDGSv"
          >
            <label
              class="sc-gKsecS dRZBKP"
              data-testid="label"
            >
              Name
            </label>
            <input
              class="sc-iBPTVF YEbKX"
              data-testid="input-name"
              name="name"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-jSguLX klDGSv"
          >
            <label
              class="sc-gKsecS dRZBKP"
              data-testid="label"
            >
              Email
            </label>
            <input
              class="sc-iBPTVF YEbKX"
              data-testid="input-email"
              name="email"
              type="email"
              value=""
            />
          </div>
          <button
            class="sc-jrAGKZ cntfQH"
            data-testid="confirm-booking"
          >
            Confirm Booking
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-idOjMB kWZquY"
    />
    <div
      class="sc-cxFKTC hsKcGb"
      data-testid="base-drawer"
    >
      <button
        class="sc-dmlpXa gYemGx"
        data-testid="close-button"
      >
        X
      </button>
      <form
        class="sc-crrtmM XATYL"
      >
        <h3
          class="sc-dQpIV fPWcU"
        >
          Confirm Your Booking
        </h3>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Name
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-name"
            name="name"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Email
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="confirm-booking"
        >
          Confirm Booking
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ConfirmYourBookingDrawer /> should render the specified email 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="sc-idOjMB kWZquY"
      />
      <div
        class="sc-cxFKTC hsKcGb"
        data-testid="base-drawer"
      >
        <button
          class="sc-dmlpXa gYemGx"
          data-testid="close-button"
        >
          X
        </button>
        <form
          class="sc-crrtmM XATYL"
        >
          <h3
            class="sc-dQpIV fPWcU"
          >
            Confirm Your Booking
          </h3>
          <div
            class="sc-jSguLX klDGSv"
          >
            <label
              class="sc-gKsecS dRZBKP"
              data-testid="label"
            >
              Name
            </label>
            <input
              class="sc-iBPTVF YEbKX"
              data-testid="input-name"
              name="name"
              type="text"
              value=""
            />
          </div>
          <div
            class="sc-jSguLX klDGSv"
          >
            <label
              class="sc-gKsecS dRZBKP"
              data-testid="label"
            >
              Email
            </label>
            <input
              class="sc-iBPTVF YEbKX"
              data-testid="input-email"
              name="email"
              type="email"
              value="test@test.com"
            />
          </div>
          <button
            class="sc-jrAGKZ cntfQH"
            data-testid="confirm-booking"
          >
            Confirm Booking
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-idOjMB kWZquY"
    />
    <div
      class="sc-cxFKTC hsKcGb"
      data-testid="base-drawer"
    >
      <button
        class="sc-dmlpXa gYemGx"
        data-testid="close-button"
      >
        X
      </button>
      <form
        class="sc-crrtmM XATYL"
      >
        <h3
          class="sc-dQpIV fPWcU"
        >
          Confirm Your Booking
        </h3>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Name
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-name"
            name="name"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Email
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-email"
            name="email"
            type="email"
            value="test@test.com"
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="confirm-booking"
        >
          Confirm Booking
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ConfirmYourBookingDrawer /> should render the specified name 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="sc-idOjMB kWZquY"
      />
      <div
        class="sc-cxFKTC hsKcGb"
        data-testid="base-drawer"
      >
        <button
          class="sc-dmlpXa gYemGx"
          data-testid="close-button"
        >
          X
        </button>
        <form
          class="sc-crrtmM XATYL"
        >
          <h3
            class="sc-dQpIV fPWcU"
          >
            Confirm Your Booking
          </h3>
          <div
            class="sc-jSguLX klDGSv"
          >
            <label
              class="sc-gKsecS dRZBKP"
              data-testid="label"
            >
              Name
            </label>
            <input
              class="sc-iBPTVF YEbKX"
              data-testid="input-name"
              name="name"
              type="text"
              value="test"
            />
          </div>
          <div
            class="sc-jSguLX klDGSv"
          >
            <label
              class="sc-gKsecS dRZBKP"
              data-testid="label"
            >
              Email
            </label>
            <input
              class="sc-iBPTVF YEbKX"
              data-testid="input-email"
              name="email"
              type="email"
              value=""
            />
          </div>
          <button
            class="sc-jrAGKZ cntfQH"
            data-testid="confirm-booking"
          >
            Confirm Booking
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-idOjMB kWZquY"
    />
    <div
      class="sc-cxFKTC hsKcGb"
      data-testid="base-drawer"
    >
      <button
        class="sc-dmlpXa gYemGx"
        data-testid="close-button"
      >
        X
      </button>
      <form
        class="sc-crrtmM XATYL"
      >
        <h3
          class="sc-dQpIV fPWcU"
        >
          Confirm Your Booking
        </h3>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Name
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-name"
            name="name"
            type="text"
            value="test"
          />
        </div>
        <div
          class="sc-jSguLX klDGSv"
        >
          <label
            class="sc-gKsecS dRZBKP"
            data-testid="label"
          >
            Email
          </label>
          <input
            class="sc-iBPTVF YEbKX"
            data-testid="input-email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <button
          class="sc-jrAGKZ cntfQH"
          data-testid="confirm-booking"
        >
          Confirm Booking
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
